third_party/move/tools/move-linter/src/model_ast_lints/equal_operands_in_pure_bin_op.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use move_compiler_v2::external_checks::ExpChecker;
+use move_model::{
+    ast::{Exp, ExpData, Operation},
+    model::FunctionEnv,
+};
+
+#[derive(Default)]
+pub struct EqualOperandsInPureBinOp;
```

Comment:
We can just call this `EqualOperandsInBinop` (correspondingly change all file names/lint names etc). All binops are pure, you presumably meant the pureness to apply to operands, but I don't think this qualification is necessary (as the check is not complete w.r.t pureness.)